### PR TITLE
OTC-279: Return duplicated payment code instead of saving them in db

### PR DESCRIPTION
**Pull request by sniedzielski**
TICKET: https://openimis.atlassian.net/browse/OTC-279

### CODE REVIEW

**Comment by sniedzielski** [comment]
@dragos-dobre That is my proposition to support duplicate error with saving this message error into original payment in "RejectedReason" and returning to GePG 7207. I marked this PR as draft because we've been waiting for Juma response about this case. 

As the next step I can implement supporting this issue: "bill id not existing (create the payment with the billid to save the error: -3 if on getCN, -4 if on GetPayment, -5 if on getRecon)" (point from description of OTC-279)

File: OpenImis.ePayment/Escape/Payment/Controllers/PaymentController.cs [hunk 1]
```diff
@@ -260,10 +260,34 @@ public async Task<IActionResult> GetPaymentChf([FromBody] gepgPmtSpInfo model)
                     GepgFileLogger.Log(billId, "Payment", reconc, env);
```
Comment (dragos-dobre) [inline OpenImis.ePayment/Escape/Payment/Controllers/PaymentController.cs:260]: _response is null so difficult to understand what we save here.

File: OpenImis.ePayment/Logic/PaymentLogic.cs [hunk 2]
```diff
@@ -212,6 +213,30 @@ public async Task<DataMessage> SavePayment(PaymentData model)
...
 
             }
+            else 
+            {
+                string rejectedReason = "";
+                switch (response.Code) 
+                {
+                    //3 - Wrong Control Number
+                    case 3:
+                        rejectedReason = payment.PrepareRejectedReason(payment.PaymentId, GepgCodeResponses.GepgResponseCodes["Bill does not exist"].ToString());
```
Comment (dragos-dobre) [inline OpenImis.ePayment/Logic/PaymentLogic.cs:223]: We cannot use GepgCodeResponses here as this is part of the generic implementation. The error message must be handled in the escape part, either in PaymentController or ImisPayment in the code between `#if CHF` and `#endif` as only this part is specific for TZ.

**Review by dragos-dobre — CHANGES_REQUESTED** [review]

**Comment by delcroip** [comment]
@dragos-dobre , @sniedzielski  Can I close this PR ?

**Pull request closed without merging**